doc/man7/life_cycle-digest.pod

```diff
@@ -0,0 +1,221 @@
+=pod
+
+=head1 NAME
+
+life_cycle-digest - The digest algorithm life-cycle
+
+=head1 DESCRIPTION
+
+All message digests (MDs) go through a number of stages in their life-cycle:
+
+=over 4
+
+=item start
+
+This state represents the MD before it has been allocated.  It is the
+starting state for any life-cycle transitions.
+
+=item newed
+
+This state represents the MD after it has been allocated.
+
+=item initialised
+
+This state represents the MD when it is set up and capable of processing
+input.
+
+=item updated
+
+This state represents the MD when it is set up and capable of processing
+additional input or generating output.
+
+=item finaled
+
+This state represents the MD when it has generated output.
+
+=item freed
+
+This state is entered when the MD is freed.  It is the terminal state
+for all life-cycle transitions.
+
+=back
+
+=head2 State Transition Diagram
+
+The usual life-cycle of a MD is illustrated:
+
+=begin man
+
+                        +-------------------+
+                        |       start       |
+                        +-------------------+
+                          |
+                          | EVP_MD_CTX_new
+                          v
+                        +-------------------+
+                     +> |       newed       | <-----------------------------+
+                     |  +-------------------+                               |
+                     |    |                                                 |
+    EVP_MD_CTX_reset |    | EVP_DigestInit                 EVP_MD_CTX_reset |
+                     |    v                                                 |
+                     |  +-------------------+                               |
+                +----+> |    initialised    | <+ EVP_DigestInit             |
```

Comment:
The crossover for the EVP_MD_CTX_reset(0 is a bit confusing since it intersects on a +

Reply:
Bloody near impossible to draw without this...

Reply:
Actually, it's not so bad...